packages/aws-rfdk/lib/deadline/lib/rq-connection.ts

```diff
@@ -114,23 +124,31 @@ export abstract class RenderQueueConnection {
         `"\${DEADLINE_PATH}/deadlinecommand" -executeScriptNoGui "${configureScriptPath}" --render-queue "${args.address}" ${dlExtraCommands.join(' ')}`,
         // Cleanup
         `rm -f "${configureScriptPath}"`,
-        'if service --status-all | grep -q "Deadline 10 Launcher"; then',
-        '  service deadline10launcher restart',
-        'fi',
       );
+      if (args.restartLauncher ?? true) {
+        host.userData.addCommands(
+          'if service --status-all | grep -q "Deadline 10 Launcher"; then',
+          '  service deadline10launcher restart',
+          'fi',
+        );
+      }
     } else if ( host.osType === OperatingSystemType.WINDOWS ) {
       host.userData.addCommands(
         '$ErrorActionPreference = "Stop"',
         '$DEADLINE_PATH = (get-item env:"DEADLINE_PATH").Value',
         `& "$DEADLINE_PATH/deadlinecommand.exe" -executeScriptNoGui "${configureScriptPath}" --render-queue "${args.address}" ${dlExtraCommands.join(' ')} 2>&1`,
         `Remove-Item -Path "${configureScriptPath}"`,
-        'If (Get-Service "deadline10launcherservice" -ErrorAction SilentlyContinue) {',
-        '  Restart-Service "deadline10launcherservice"',
-        '} Else {',
-        '  & "$DEADLINE_PATH/deadlinelauncher.exe" -shutdownall 2>&1',
-        '  & "$DEADLINE_PATH/deadlinelauncher.exe" 2>&1',
-        '}',
       );
+      if (args.restartLauncher ?? true) {
+        host.userData.addCommands(
+          'If (Get-Service "deadline10launcherservice" -ErrorAction SilentlyContinue) {',
+          '  Restart-Service "deadline10launcherservice"',
+          '} Else {',
+          '  & "$DEADLINE_PATH/deadlinelauncher.exe" -shutdownall 2>&1',
+          '  & "$DEADLINE_PATH/deadlinelauncher.exe" 2>&1',
```

Comment:
I see below that you added the missing `-nogui` options. Add it here as well?

Reply:
Good catch - fixed